python-package/lightgbm/compat.py

```diff
@@ -75,9 +75,9 @@ def __init__(self, *args, **kwargs):
     from sklearn.utils.validation import assert_all_finite, check_array, check_X_y
     try:
         from sklearn.exceptions import NotFittedError
-        from sklearn.model_selection import GroupKFold, StratifiedKFold
+        from sklearn.model_selection import BaseCrossValidator, GroupKFold, StratifiedKFold
```

Comment:
`sklearn.model_selection.BaseCrossValidator` has been in `scikit-learn` for at least 7 years, so I don't think this will cause any portability issues.

https://github.com/scikit-learn/scikit-learn/blame/159cb46c5672b0ba87d0ba80c6b80ec1aa5fda32/sklearn/model_selection/__init__.py#L3